core/src/main/java/org/apache/gravitino/hook/FilesetHookDispatcher.java

```diff
@@ -103,8 +104,11 @@ public Fileset alterFileset(NameIdentifier ident, FilesetChange... changes)
 
   @Override
   public boolean dropFileset(NameIdentifier ident) {
+    List<String> locations =
+        AuthorizationUtils.getMetadataObjectLocation(ident, Entity.EntityType.FILESET);
     boolean dropped = dispatcher.dropFileset(ident);
-    AuthorizationUtils.authorizationPluginRemovePrivileges(ident, Entity.EntityType.FILESET);
+    AuthorizationUtils.authorizationPluginRemovePrivileges(
+        ident, Entity.EntityType.FILESET, locations);
     return dropped;
```

Comment:
Why do we perform the authorization logic here, can we move them to the plugin?

Reply:
We need to get metadata location before dropping.